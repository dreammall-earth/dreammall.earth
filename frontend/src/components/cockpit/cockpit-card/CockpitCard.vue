<template>
  <section class="cockpit-card">
    <div v-if="slots.header" class="header">
      <slot name="header"></slot>
    </div>
    <hr v-if="slots.header" class="hr" />
    <slot></slot>
  </section>
</template>

<script setup lang="ts">
import { useSlots } from 'vue'

const slots = useSlots()
</script>

<style scoped lang="scss">
@use 'sass:map';
@import 'vuetify/lib/styles/settings/_variables';

.cockpit-card {
  --padding: 20px;
  --card-width: 100%;

  max-width: var(--card-width, 100%);
  padding: var(--padding);
  background-color: var(--v-sidebar-background);
  border-radius: 20px;
}

.header {
  display: flex;
  flex-flow: column;
  justify-content: center;
  height: 28px;
  text-wrap: nowrap;
}

.header:deep(h2) {
  font-size: 16px;
}

.hr {
  margin-block: 15px;
  margin-inline: calc(-1 * var(--padding));
  border-color: rgb(var(--v-theme-background));
  border-style: solid;
  border-width: 0.5px;
}
</style>
