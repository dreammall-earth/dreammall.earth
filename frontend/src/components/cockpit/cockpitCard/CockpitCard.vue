<template>
  <section class="cockpit-card" :class="{ narrow: props.narrow }">
    <div v-if="slots.header" class="header">
      <slot name="header"></slot>
    </div>
    <hr v-if="slots.header" class="hr" />
    <slot></slot>
  </section>
</template>

<script setup lang="ts">
import { useSlots } from 'vue'

const slots = useSlots()

const props = defineProps<{
  narrow?: boolean
}>()
</script>

<style scoped lang="scss">
.cockpit-card {
  --padding: 20px;
  --cockpit-card-height: 323px;

  max-height: var(--cockpit-card-height);
  padding: var(--padding);
  background-color: var(--v-sidebar-background);
  border-radius: 20px;
}

.header:deep(h2) {
  font-size: 16px;
}

.hr {
  margin-block: 15px;
  margin-inline: calc(-1 * var(--padding));
  border-color: rgb(var(--v-theme-background));
  border-style: solid;
  border-width: 0.5px;
}

.narrow {
  max-width: 335px;
}
</style>
