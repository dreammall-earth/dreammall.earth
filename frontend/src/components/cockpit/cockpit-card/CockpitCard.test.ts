import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import CockpitCard from './CockpitCard.vue'

describe('CockpitCard', () => {
  const Wrapper = ({
    narrow = false,
    headerSlot = '<h2>Header</h2>',
    defaultSlot = '<div>Content</div>',
  } = {}) =>
    mount(CockpitCard, { props: { narrow }, slots: { default: defaultSlot, header: headerSlot } })

  it('renders', () => {
    const wrapper = Wrapper()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders narrow', () => {
    const wrapper = Wrapper({ narrow: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders with empty header slot', () => {
    const wrapper = Wrapper({ headerSlot: undefined })
    expect(wrapper.element).toMatchSnapshot()
  })
})
