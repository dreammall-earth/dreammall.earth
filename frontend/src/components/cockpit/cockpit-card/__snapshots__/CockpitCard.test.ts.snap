// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CockpitCard > renders 1`] = `
<section
  class="cockpit-card"
  data-v-1451ba78=""
  narrow="false"
>
  <div
    class="header"
    data-v-1451ba78=""
  >
    
    <h2>
      Header
    </h2>
    
  </div>
  <hr
    class="hr"
    data-v-1451ba78=""
  />
  
  <div>
    Content
  </div>
  
</section>
`;

exports[`CockpitCard > renders narrow 1`] = `
<section
  class="cockpit-card"
  data-v-1451ba78=""
  narrow="true"
>
  <div
    class="header"
    data-v-1451ba78=""
  >
    
    <h2>
      Header
    </h2>
    
  </div>
  <hr
    class="hr"
    data-v-1451ba78=""
  />
  
  <div>
    Content
  </div>
  
</section>
`;

exports[`CockpitCard > renders with empty header slot 1`] = `
<section
  class="cockpit-card"
  data-v-1451ba78=""
  narrow="false"
>
  <div
    class="header"
    data-v-1451ba78=""
  >
    
    <h2>
      Header
    </h2>
    
  </div>
  <hr
    class="hr"
    data-v-1451ba78=""
  />
  
  <div>
    Content
  </div>
  
</section>
`;
